xmtp_mls/src/storage/encrypted_store/user_preferences.rs

```diff
@@ -5,30 +5,53 @@ use super::{
     DbConnection,
 };
 use diesel::prelude::*;
+use rand::{rngs::OsRng, RngCore};
 
-#[derive(Identifiable, Insertable, Queryable, Debug, Clone, PartialEq, Eq, Default)]
+#[derive(Identifiable, Insertable, Queryable, Debug, Clone, PartialEq, Eq)]
 #[diesel(table_name = user_preferences)]
 #[diesel(primary_key(id))]
 pub struct StoredUserPreferences {
     /// Primary key - latest key is the "current" preference
     pub id: Option<i32>,
     /// Randomly generated hmac key root
-    pub hmac_key: Option<Vec<u8>>,
+    pub hmac_key: Vec<u8>,
 }
+
+impl Default for StoredUserPreferences {
+    fn default() -> Self {
+        let mut hmac_key = vec![0; 32];
+        OsRng.fill_bytes(&mut hmac_key);
+
+        Self { id: None, hmac_key }
+    }
+}
+
 impl_store!(StoredUserPreferences, user_preferences);
 
 impl StoredUserPreferences {
     pub fn load(conn: &DbConnection) -> Result<Self, StorageError> {
         let query = dsl::user_preferences.order(dsl::id.desc()).limit(1);
         let mut result = conn.raw_query(|conn| query.load::<StoredUserPreferences>(conn))?;
 
-        Ok(result.pop().unwrap_or_default())
+        let result = match result.pop() {
+            Some(result) => result,
+            None => {
+                // Create a default and store it.
+                let result = Self::default();
+                // TODO: emit an hmac key update event here.
```

Comment:
do we need this in this PR or is it OK to exclude for now?/ is it for streams

Reply:
It's the next step. I have a TODO just to mark it to make it easy in the next PR :)